### PR TITLE
More strongly typed telemetry

### PR DESCRIPTION
**Pull request by lramos15**
Just converting some more of our telemetry into typescript annotations rather than GDPR comments. This doesn't change any runtime functionality and our telemetry reports are the same, it's just refactoring.

### CODE REVIEW

**Comment by lramos15** [comment]
1. walkThroughSnippetInteraction
2. commandExecuted
3. workspace.stats
4. update:downloaded
5. update:notAvailable
6. galleryService: query
7. taskService.template
8. workspaceLoad
9. Experiments
10. shimming.open
11. shimming.open.call.no.Forward

File: src/vs/platform/update/electron-main/updateService.snap.ts [hunk 1]
```diff
@@ -160,28 +160,20 @@ export class SnapUpdateService extends AbstractUpdateService2 {
 	protected doCheckForUpdates(context: any): void {
 		this.setState(State.CheckingForUpdates(context));
 
+		type UpdateNotAvailableClassifcation = {
```
Comment (sbatten) [inline src/vs/platform/update/electron-main/updateService.snap.ts]: this type is declared many times

Comment (lramos15) [reply src/vs/platform/update/electron-main/updateService.snap.ts]: Yes it's a small type and I didn't know of a good common place to put it.

Comment (sbatten) [reply src/vs/platform/update/electron-main/updateService.snap.ts]: https://github.com/Microsoft/vscode/blob/fe3e0cdcdeb4a7f5bc2745bda73564ba4ddc75b7/src/vs/platform/update/electron-main/abstractUpdateService.ts

Comment (sbatten) [reply src/vs/platform/update/electron-main/updateService.snap.ts]: or https://github.com/Microsoft/vscode/blob/653280d133bf36f182189ded3506d3603d09fcd7/src/vs/platform/update/common/update.ts

**Pull request merged**